filesmanager/filesmanager.py

```diff
@@ -505,8 +505,21 @@ def generate_content_path(self, base_path, name=None):
             str: The new file name.
         """
         if base_path in self.content_paths:
-            base_path = f"{base_path} ({len(self.content_paths)})"
-            name = f"{name} ({len(self.content_paths)})"
+            counter = 1
+            old_name = name
+            extension = old_name.split(".")[-1]
+            base_name = old_name.replace(f".{extension}", "")
+            base_file_path = base_path.split("/")
+            del base_file_path[-1]
+            base_file_path = "/".join(base_file_path)
+            while True:
+                new_path = f"{base_file_path}/{base_name} ({counter}).{extension}"
+                counter+=1
+                if new_path not in self.content_paths:
+                    break
+
+            base_path = new_path
+            name = base_path.split("/")[-1]
```

Comment:
Since at the end we end up replacing name:
```suggestion
            extension = name.split(".")[-1]
            base_name = name.replace(f".{extension}", "")
            base_file_path = base_path.split("/")
            del base_file_path[-1]
            base_file_path = "/".join(base_file_path)
            while True:
                new_path = f"{base_file_path}/{base_name} ({counter}).{extension}"
                counter += 1
                if new_path not in self.content_paths:
                    break

            base_path = new_path
            name = base_path.split("/")[-1]
```